test_authproxy.py

```diff
@@ -0,0 +1,94 @@
+import http
+import os
+import threading
+import urllib
+from http.server import BaseHTTPRequestHandler
+from urllib.parse import urlparse
+
+import pytest
+import responses
+
+CONFIG_VARS = {
+    'DEBUG': 'DEBUG',
+    'LIQUID_CLIENT_ID': 'LIQUID_CLIENT_ID',
+    'LIQUID_CLIENT_SECRET': 'LIQUID_CLIENT_SECRET',
+    'SECRET_KEY': 'SECRET_KEY',
+    'LIQUID_PUBLIC_URL': 'http://liquid_public_url',
+    'CONSUL_URL': 'http://consul_url',
+    'LIQUID_CORE_SERVICE': 'liquid_core_service',
+    'UPSTREAM_SERVICE': 'upstream_service',
+}
+
+
+class handler_class(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html')
+        self.send_header('charset', 'utf-8')
+        self.end_headers()
+        self.wfile.write(urllib.parse.unquote(self.path).encode('utf-8'))
+        return
+
+
+@pytest.fixture
+def my_fixture():
+    handler = handler_class
+    httpd = http.server.HTTPServer(('', 8000), handler)
+    http_thread = threading.Thread(target=httpd.serve_forever)
+    http_thread.start()
+    yield
+    httpd.shutdown()
+
+
+@responses.activate
+def test_special_characters(my_fixture):
+    for key in CONFIG_VARS:
+        os.environ[key] = CONFIG_VARS[key]
+
+    mock_requests()
+
+    from authproxy import app
+
+    with app.test_client() as client:
+        with client.session_transaction() as sess:
+            sess['access_token'] = 'access_token'
+
+        rv = client.get('/Überraschung')
+        assert bytes('/Überraschung', 'utf-8') == rv.data
+
+
+def mock_requests():
+    responses.add(
+        responses.GET,
+        f'{CONFIG_VARS["CONSUL_URL"]}/v1//health/checks/{CONFIG_VARS["LIQUID_CORE_SERVICE"]}',
+        json=[{'ServiceID': 1, 'Status': 'passing'}],
+        status=200,
+    )
+
+    responses.add(
+        responses.GET,
+        f'{CONFIG_VARS["CONSUL_URL"]}/v1/catalog/service/{CONFIG_VARS["LIQUID_CORE_SERVICE"]}',
```

Comment:
You're getting into really long lines. I suggest you extract `consul_url` and `liquid_core_service` as variables. Also, it might be nicer to extract the url and the json payload, maybe the `responses.add` call fits into a single line? Also, `status=200` is [the default](https://github.com/getsentry/responses/blob/master/responses.py#L339), you can skip it. So is `json=` as the 3rd positional argument.